core/state_processor.go

```diff
@@ -86,6 +86,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 // for the transaction, gas used and an error if the transaction failed,
 // indicating the block was invalid.
 func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, error) {
+	if !config.IsYoloV2(header.Number) && tx.Type() != types.LegacyTxId {
+		return nil, ErrTxTypeNotSupported
+	}
```

Comment:
Other than rejecting non-legacy txs before yolov2 (or whatever we later call it), don't we also need to
- Verify that the type is exactly the AccessList-type, if access lists are provided? (so we reject legacy-txs with access-lists, and foobar-typed txs in general)

Or is that done somewhere else? This cannot rely on txpool verifications, since it may come in the form of a maliciously mined block

Reply:
Whenever any RLP bytes are decoded, it will determine if the transaction type is "known": https://github.com/ethereum/go-ethereum/pull/21502/files#diff-8fb609018eccff10931645347515b8ecc194992cc4ef9e84ebdea87682dd50c3R123

So it would be impossible for any other transaction type to be accepted as it will always error out. If you would prefer explicit checks in the state processor (or elsewhere?), I can add those. The legacy transaction cannot have an access list, because it's return is value is constant: https://github.com/ethereum/go-ethereum/pull/21502/files#diff-366ae3d3949ab2942dd8810a703bf0531f5ade9b7aa1d144e60aac3c2a7cf689R78